mono/utils/mono-dl.h

```diff
@@ -22,8 +23,9 @@
 
 typedef struct {
 	void *handle;
-	int main_module;
-
+	char *full_name;
+	// Only used in native-library.c
+	MonoRefCount ref;
```

Comment:
I know you don't have a destructor on `mono_refcount_init`, but if we ever add one, it expects the `ref` field to be first in the struct.

Reply:
Also would be good to mention that you have to hold the native library lock to increment/decrement this count